pkg/standalone-utils/contracts/BatchRelayer.sol

```diff
@@ -74,6 +75,21 @@ contract BatchRelayer {
         return getVault().batchSwap(IVault.SwapKind.GIVEN_IN, swaps, assets, funds, limits, deadline);
     }
 
+    function joinAndStake(
+        bytes32 poolId,
+        address payable recipient,
+        IVault.JoinPoolRequest calldata joinPoolRequest,
+        IMultiRewards stakingContract
+    ) external {
+        getVault().joinPool(poolId, msg.sender, address(this), joinPoolRequest);
+
+        IERC20 bpt = IERC20(_getPoolAddress(poolId));
+        uint256 bptAmount = bpt.balanceOf(address(this));
+
+        bpt.approve(address(stakingContract), bptAmount);
```

Comment:
```suggestion
        // If necessary, give Vault allowance to take BPT
        if (bpt.allowance(address(this), address(getVault())) < bptAmount) {
            bpt.approve(address(getVault()), type(uint256).max);
        }
        
```

Reply:
This will reduce gas costs by about 20k after the first usage for each pool.

An infinite approval is safe here as the relayer will never hold a balance of BPT as it is instantly staked.